test/arcade/patch/agent/cell/PatchCellCARTTest.java

```diff
@@ -0,0 +1,235 @@
+        parameters = spy(new Parameters(new MiniBox(), null, null));
+        location = mock(PatchLocation.class);
+
+        doReturn(0.0).when(parameters).getDouble(any(String.class));
+        doReturn(0).when(parameters).getInt(any(String.class));
+
+        int id = 1;
+        int parentId = 1;
+        int pop = 1;
+        int age = randomIntBetween(1, 120950);
+        int divisions = 10;
+        double volume = randomDoubleBetween(100, 200);
+        double height = randomDoubleBetween(4, 10);
+        double criticalVolume = randomDoubleBetween(100, 200);
+        double criticalHeight = randomDoubleBetween(4, 10);
+        PatchEnums.State state = PatchEnums.State.UNDEFINED;
+        ;
+
+        container =
+                new PatchCellContainer(
+                        id,
+                        parentId,
+                        pop,
+                        age,
+                        divisions,
+                        state,
+                        volume,
+                        height,
+                        criticalVolume,
+                        criticalHeight);
+
+        when(parameters.getDouble("HETEROGENEITY")).thenReturn(0.0);
+        when(parameters.getDouble("ENERGY_THRESHOLD")).thenReturn(1.0);
+
+        when(parameters.getDouble("NECROTIC_FRACTION"))
+                .thenReturn(randomIntBetween(40, 100) / 100.0);
+        when(parameters.getDouble("EXHAU_FRAC")).thenReturn(randomIntBetween(40, 100) / 100.0);
+        when(parameters.getDouble("SENESCENT_FRACTION"))
+                .thenReturn(randomIntBetween(40, 100) / 100.0);
+        when(parameters.getDouble("ANERGIC_FRACTION"))
+                .thenReturn(randomIntBetween(40, 100) / 100.0);
+        when(parameters.getDouble("PROLIFERATIVE_FRACTION"))
+                .thenReturn(randomIntBetween(40, 100) / 100.0);
+        when(parameters.getInt("SELF_RECEPTORS")).thenReturn(randomIntBetween(100, 200));
+        when(parameters.getDouble("SEARCH_ABILITY")).thenReturn(1.0);
+        when(parameters.getDouble("CAR_AFFINITY")).thenReturn(10 * Math.pow(10, -7));
+        when(parameters.getDouble("CAR_ALPHA")).thenReturn(3.0);
+        when(parameters.getDouble("CAR_BETA")).thenReturn(0.01);
+        when(parameters.getDouble("SELF_RECEPTOR_AFFINITY")).thenReturn(7.8E-6);
+        when(parameters.getDouble("SELF_ALPHA")).thenReturn(3.0);
+        when(parameters.getDouble("SELF_BETA")).thenReturn(0.02);
+        when(parameters.getDouble("CONTACT_FRAC")).thenReturn(7.8E-6);
+        when(parameters.getInt("MAX_ANTIGEN_BINDING")).thenReturn(10);
+        when(parameters.getInt("CARS")).thenReturn(50000);
+
+        when(parameters.getInt("APOPTOSIS_AGE")).thenReturn(120960);
+        when(parameters.getInt("MAX_DENSITY")).thenReturn(54);
+
+        patchCellCART = new PatchCellMock(container, location, parameters);
+    }
+
+    @Test
+    public void testSetAntigenFlag() {
```

Comment:
the general style of tests names (these are obviously tests, so you should drop that) - 

`{method under test}_{system state}_ {expected result}`, see other tests in other patch classes